crypto/fipsmodule/cipher/cipher.c

```diff
@@ -480,12 +493,17 @@ int EVP_DecryptFinal_ex(EVP_CIPHER_CTX *ctx, unsigned char *out, int *out_len) {
     *out_len = 0;
   }
 
+  AES_verify_service_indicator_with_ctx(ctx);
   return 1;
 }
 
 int EVP_Cipher(EVP_CIPHER_CTX *ctx, uint8_t *out, const uint8_t *in,
                size_t in_len) {
-  return ctx->cipher->cipher(ctx, out, in, in_len);
+  int ret = ctx->cipher->cipher(ctx, out, in, in_len);
+  if(ret == 1) {
```

Comment:
> For stream ciphers, the stream is advanced past the bytes
// used. It returns one on success and zero otherwise, unless |EVP_CIPHER_flags|
// has |EVP_CIPH_FLAG_CUSTOM_CIPHER| set. Then it returns the number of bytes
// written or -1 on error.

I guess the more precise statement is that the return value depends on the actual cipher... Since you don't know that here, well, I don't think you can just check `== 1`.

You can make a bigger argument that we only validate ciphers that doesn't set `EVP_CIPH_FLAG_CUSTOM_CIPHER`.  But there is no need to make that argument imo.